### PR TITLE
Integrate LavaDome private key protection

### PR DESCRIPTION
**Pull request by weizman**
## **Description**

New [LavaDome](https://github.com/LavaMoat/LavaDome) tool can help to better hide the private key from potential malicious code running in the app when the user asks to see it.

## **Related issues**

None

## **Manual testing steps**

This PR changes the "show private key" dialog, not in terms of appearance, and not in terms of interaction but only in terms of "behind the scenes" security, so nothing should change in terms of manual interaction with this feature - that is the most important thing to check here QA-wise.

One thing that is changed, is that the private key text is no longer selectable as it was before, and now the only way to copy it is to use the copy-to-clipboard button.

That is on purpose, allowing selection of sensitive information such as the private key is unsafe, and disablement of selection capabilities of sensitive info is done intentionally to enhance security.

### **Before**

Screenshot 2024-01-10 at 22 43 55

### **After**

key in image isn't real

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by weizman** [comment]
Once https://github.com/LavaMoat/LavaDome/pull/18 is merged and published, eab7582 should work too

File: test/e2e/tests/account-details.spec.js [hunk 1]
```diff
@@ -40,7 +41,7 @@ describe('Show account details', function () {
       '[data-testid="account-details-key"]',
     );
     const key = await keyContainer.getText();
-    return key;
+    return LavaDomeDebug.stripDistractionFromText(key);
```
Comment (weizman) [inline test/e2e/tests/account-details.spec.js:44]: see [stripDistractionFromText](https://github.com/LavaMoat/LavaDome?tab=readme-ov-file#stripdistractionfromtext)

File: ui/components/multichain/account-details/account-details-key.js [hunk 2]
```diff
@@ -50,7 +53,7 @@ export const AccountDetailsKey = ({ accountName, onClose, privateKey }) => {
           variant={TextVariant.bodySm}
           style={{ wordBreak: 'break-word' }}
         >
-          {privateKey}
+          <LavaDomeReact unsafeOpenModeShadow={inTest} text={privateKey} />
```
Comment (weizman) [inline ui/components/multichain/account-details/account-details-key.js:56]: see `@unsafeOpenModeShadow` at [getTextByRoot](https://github.com/LavaMoat/LavaDome?tab=readme-ov-file#gettextbyroot)

File: ui/components/multichain/account-details/account-details.test.js [hunk 3]
```diff
@@ -93,16 +94,22 @@ describe('AccountDetails', () => {
...
+          onClose={jest.fn()}
+          privateKey={samplePrivateKey}
+        />
+      </div>,
     );
 
-    expect(queryByText(samplePrivateKey)).toBeInTheDocument();
+    const hook = queryByText(textHook);
+    const rootNode = hook.parentElement.querySelector('span');
+    expect(LavaDomeDebug.getTextByRoot(rootNode)).toContain(samplePrivateKey);
```
Comment (weizman) [inline ui/components/multichain/account-details/account-details.test.js:112]: see [getTextByRoot](https://github.com/LavaMoat/LavaDome?tab=readme-ov-file#gettextbyroot)

File: ui/components/multichain/account-details/account-details-key.js (hunk 2 above)
Comment (weizman) [reply ui/components/multichain/account-details/account-details-key.js:56]: see [react](https://github.com/LavaMoat/LavaDome?tab=readme-ov-file#react) package usage

**Comment by chloeYue** [comment]
QA passed, all following scenarios have been tested:
- user can still reveal their private key
- user can still copy their private key using the copy button
- user cannot copy their private key any other way (i.e. the can't select the text and copy it)
- private key cannot be seen if the html is inspected with the browser tools

**Review by darkwing — DISMISSED** [review]

**Review by danjm — COMMENTED** [review]
The code changes look good to me. I will approve when we get another QA pass

**Comment by danjm** [comment]
actually, I will QA this myself

**Comment by danjm** [comment]
>QA passed, all following scenarios have been tested:

>user can still reveal their private key
>user can still copy their private key using the copy button
>user cannot copy their private key any other way (i.e. the can't select the text and copy it)
>private key cannot be seen if the html is inspected with the browser tools

These tests continue to pass, and I can also see the implemented message upon clicking the private key

Screenshot from 2024-01-31 14-03-13

**Review by danjm — APPROVED** [review]

File: ui/components/multichain/account-details/account-details-key.js [hunk 4]
```diff
@@ -50,13 +56,22 @@ export const AccountDetailsKey = ({ accountName, onClose, privateKey }) => {
           variant={TextVariant.bodySm}
           style={{ wordBreak: 'break-word' }}
         >
-          {privateKey}
+          <span onClick={() => setShowDisableSelectWarn(true)}>
```
Comment (georgewrmarshall) [inline ui/components/multichain/account-details/account-details-key.js:59]: nit: could we reduce html bloat and increase the click area of the text slightly by moving the onClick to the wrapping `Text` component?

File: ui/components/multichain/account-details/account-details-key.js [hunk 5]
```diff
@@ -50,13 +56,22 @@ export const AccountDetailsKey = ({ accountName, onClose, privateKey }) => {
...
+            <LavaDomeReact unsafeOpenModeShadow={inTest} text={privateKey} />
+          </span>
         </Text>
         <ButtonIcon
-          onClick={() => handlePrivateKeyCopy(privateKey)}
+          onClick={() =>
+            setShowDisableSelectWarn(false) || handlePrivateKeyCopy(privateKey)
+          }
```
Comment (georgewrmarshall) [inline ui/components/multichain/account-details/account-details-key.js:66]: non-blocking: To improve the accessibility here we could add an `ariaLabel={t('copyPrivateKey')}` which would need to be created in `messages.json`. ariaLabel should be a required prop for this component

**Review by georgewrmarshall — APPROVED** [review]
LGTM! Tested and it works well. Nice use of component-library components. Left 2 non-blocking comments

File: ui/components/multichain/account-details/account-details-key.js (hunk 4 above)
Comment (weizman) [reply ui/components/multichain/account-details/account-details-key.js:59]: https://github.com/MetaMask/metamask-extension/pull/22765/commits/d531a6715697b8ef22bfbd825a22804a2c3874ea

File: ui/components/multichain/account-details/account-details-key.js (hunk 5 above)
Comment (weizman) [reply ui/components/multichain/account-details/account-details-key.js:66]: https://github.com/MetaMask/metamask-extension/pull/22765/commits/7ceba331e977edf5d30c0bfa6e97baa04fb2b373

**Pull request merged**